src/layers/core/geojson-layer/geojson-layer.js

```diff
@@ -110,8 +115,8 @@ export default class GeoJsonLayer extends Layer {
 
     // Override user's onHover and onClick props
     const handlers = {
-      onHover: this._onHoverSublayer.bind(this),
-      onClick: noop
+      onHover: this._onPickSublayer.bind(this, 'hover'),
```

Comment:
Nit: I'd drop the `handlers` object and copy these lines into each layer below. I think the readability would be better (Composition using `Object.assign` doesn't read as well as `...`), but could just be my taste.

We could still bind them here before creating the objects.